use_cases/init-process-for-fluent-bit/README.md

```diff
@@ -0,0 +1,227 @@
+                   {
+                       "name": "aws_fluent_bit_init_file_1",
+                       "value": "/ecs/s3.conf"
+                   }
+               ]
+   ```
+
+   **We support two kinds of config files:**
+
+   * Config files which stored in your S3 bucket
+     You need to set the name of the env var using prefix `aws_fluent_bit_init_s3_`, the number after this prefix cannot be repeated, and set the `ARN` of your config file as the value of the env var.
+
+   * Config files which from the image
+
+     * Use [our built-in config files](https://github.com/aws/aws-for-fluent-bit/tree/mainline/ecs)
+     
+       You need to set the name of the env var using prefix `aws_fluent_bit_init_file_`, the number after this prefix cannot be repeated, and set the `Path` of your config file as the value of the env var.
+     
+     * Build a custom image
+     
+       You can build a custom image using image with `init` tag as the base and add your own configs. For more details please see [this example](https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/config-file-type-file).
+     
+
+3. Set up your ECS Task role
+
+   If you specify the config file from your S3 bucket, the init process will download it from your bucket. So please add the following permissions: 
+
+   ``` 
+   {
+       "Version": "2012-10-17",
+       "Statement": [
+           {
+               "Effect": "Allow",
+               "Action": [
+                   "s3:GetObject",
+                   "s3:GetBucketLocation"
+               ],
+               "Resource": "*"
+           }
+       ]
+   }
+   ```
+
+
+
+### How to use ECS Task Metadata in Fluent Bit config?
+
+The init process injects ECS Task Metadata into the Fluent Bit container as environment variables:
+
+```
+AWS_REGION / ECS_LAUNCH_TYPE / ECS_CLUSTER / ECS_FAMILY
+ECS_TASK_ARN / ECS_TASK_ID / ECS_REVISION / ECS_TASK_DEFINITION
+```
+
+You can use them as env vars directly in the Fluent Bit config.
+
+**Example:**
+
+```
+[OUTPUT]
+    Name   		cloudwatch_logs
+    Match  		*
+    region 		${AWS_REGION}
+```
+
+
+
+### How init process works?
+
+1. The init process will request [ECS Task Metadata](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html). Get valuable parts from the responses and set them using `export` as environment variables.
+
+2. After you set the ARN or Path of the config files as environment variables in the ECS Task Definition, the init process will collect all config files which you specified (download files which come from S3 if needed).
+
+3. The init process will process these config files one by one, use @INCLUDE keyword to add config files to the main config file. And the init process will check if each config file is a parser config. If it is a parser config, change the original Fluent Bit command, add -R to specify that parser. For more details, please see the example below.
```

Comment:
I think you need another step for the fact that you also @INCLUDE the firelens generated config file: https://github.com/aws-samples/amazon-ecs-firelens-under-the-hood/blob/mainline/generated-configs/fluent-bit/generated_by_firelens.conf

https://aws.amazon.com/blogs/containers/under-the-hood-firelens-for-amazon-ecs-tasks/